datatree/treenode.py

```diff
@@ -236,26 +235,53 @@ def _post_attach_children(self: Tree, children: Mapping[str, Tree]) -> None:
         """Method call after attaching `children`."""
         pass
 
-    def iter_lineage(self: Tree) -> Iterator[Tree]:
+    def _iter_parents(self: Tree) -> Iterator[Tree]:
```

Comment:
I kept the private `_iter_parents` method as it is useful to insulate the "low-level" logic (while + yield) from the `parents` method that just have to consume the iterator into a tuple. I could not find a way of not keeping this intermediate private method that would have been more elegant